### DISCUSSION TITLE
Importing Accel World vs Sword Art Online into Blender

### DISCUSSION THREAD

**Discussion by Erupuran** [rokups/rbfx#629] [General]
I would like you to create a script to import and export the Accel World vs Sword Art Online character model .mdl file to Blender.
Or please tell me how to swap character models using Note Pad etc.

**Comment by gleblebedev** [comment]
I don't think the https://store.steampowered.com/app/607880/Accel_World_VS_Sword_Art_Online_Deluxe_Edition/ game is made with rbfx engine. Do you know what game engine is it based on?

**Reply by Erupuran** [reply]
What information about a game can I find out about the game engine?
At a later date, I will show you the name of the company involved in the production of the game, which will be displayed when  start the game, so please wait.

**Reply by JTippetts** [reply]
Given that Art Dink's history shows titles going back to the late 80s, I suspect they use in house engines and frameworks, so getting actual information about what they use and what formats they have developed is likely to be tough. I doubt it's worth the time and effort to reverse engineer anything.

**Reply by Erupuran** [reply]
The companies that appear when start the game are ``HAVOK'' and ``ARTDINK.''
Is the game engine "ARTDINK"?
However, on a site called XenTax, someone created a script to import Accel World vs Sword Art Online's character model.mdl into Blender.
But it is only possible to import, it is not possible to export as .mdl file.
I wish I knew how to export it.

**Reply by JTippetts** [reply]
Art dink is the name of the developer. https://en.wikipedia.org/wiki/Artdink Again, most likely they use an in house engine. If there is a script for importing to Blender, then that goes a long way toward solving your problem as you can study the source code of the script to see what it is doing and how the data it is importing is formatted.

**Reply by Erupuran** [reply]
Someone said that a .mdl model file is an abbreviation for "model", not an extension indicating the file format.
Can model files that are not in a fixed format like this be edited only with software that can add add-ons such as Blender?
There's no need to stick to Blender.

**Reply by JTippetts** [reply]
.mdl is just an extension. They could have used literally any extension for their model format and it would have changed nothing. Some extensions are significant and mean something, like zip, but even with those there are no guarantees that a .zip file is actually  zip archive. So worry less about the extension because it means nothing other than that the dev picked it for their models. Worry more about the format of the data in the file, which if it is a custom format will need to be figured out. Which is why the existence of a blender importer script is nice because you can look at the script to see how the data in the file is laid out, and use that knowledge to reverse the process.

**Reply by Erupuran** [reply]
How can I find out if a file is in custom format?

**Reply by rokups** [reply]
This is a reverse engineering question. We wont be able to help with that :)

**Reply by JTippetts** [reply]
You use Google to find out, mainly. If the file extension were something like .md2, Google would come back and tell you MD2 is a file format used by Quake 2, and borrowed by numerous other projects. So if you had .md2 files you could try opening them with any .md2 importer and see if it worked. Googling for the .MDL extension tells you that a .MDL file could be a MathWorks Simulink file. The spec for this file format says it is a plain-text (meaning you can open it in a text editor and read it plainly). However, Google also says that Valve has used the .MDL extension for Half Life model files. So you could try opening the files with a Half Life model viewer.

However, there is no guarantee that a .MDL file is a MathWorks or Half Life file, because there is no standard at play here. Anybody can use any extension for their model files. So if you can't open the file with a Half Life viewer, or with MathWorks Simulink, or with anything else that Google tells you it might be, then you are dealing with somebody's custom format and it will have to be reverse engineered. We can't really help you with that, as rk said.

Once again, you said that there exists a Blender script for importing the files to Blender. Twice I've recommended you look at it, but it seems you haven't listened. So for a third and final time: LOOK AT THE BLENDER SCRIPT. Any script that can import data from a (apparently custom) file format like this can be used as a tool for you to puzzle out the file format and be able to write another script for exporting to that same format.

**Reply by Erupuran** [reply]
Thank you for your detailed explanation.
When I searched on Google myself, I found words such as MathWorks Simulink and Half Life.
However, Half Life is a specific game software, and I tried it, but it still failed.
I haven't tried MathWorks Simulink much yet, so I'll try various things.

I'll post a screenshot of the Blender script here at a later date.
There is a high possibility that it will be a hint of something.

**Reply by Erupuran** [reply]
![Blender  C__Users_ryosu_Desktop_その他_一時保管フォルダ_Blender249 Accel World VS Sword Art Online PC mdl anm 2017-12-04 _Blender249 Accel World VS Sword Art Online PC mdl anm 2017-12-04 _Blender249 blend  2023_11_14 1_11_30](https://github.com/rbfx/rbfx/assets/148245940/67e19ac7-4f6f-4596-8ace-e5242ad6c7f1)
![Blender  C__Users_ryosu_Desktop_その他_一時保管フォルダ_Blender249 Accel World VS Sword Art Online PC mdl anm 2017-12-04 _Blender249 Accel World VS Sword Art Online PC mdl anm 2017-12-04 _Blender249 blend  2023_11_14 1_12_17](https://github.com/rbfx/rbfx/assets/148245940/7f04c6d5-ab81-49e2-8e6b-e5c119ef9300)
![Blender  C__Users_ryosu_Desktop_その他_一時保管フォルダ_Blender249 Accel World VS Sword Art Online PC mdl anm 2017-12-04 _Blender249 Accel World VS Sword Art Online PC mdl anm 2017-12-04 _Blender249 blend  2023_11_14 1_12_54](https://github.com/rbfx/rbfx/assets/148245940/824249e9-3517-4b77-bc26-dc2f98f5de04)
![Blender  C__Users_ryosu_Desktop_その他_一時保管フォルダ_Blender249 Accel World VS Sword Art Online PC mdl anm 2017-12-04 _Blender249 Accel World VS Sword Art Online PC mdl anm 2017-12-04 _Blender249 blend  2023_11_14 1_14_17](https://github.com/rbfx/rbfx/assets/148245940/81c00913-01d7-4ff5-ac42-8fea4e19c7b2)
![Blender  C__Users_ryosu_Desktop_その他_一時保管フォルダ_Blender249 Accel World VS Sword Art Online PC mdl anm 2017-12-04 _Blender249 Accel World VS Sword Art Online PC mdl anm 2017-12-04 _Blender249 blend  2023_11_14 1_10_35](https://github.com/rbfx/rbfx/assets/148245940/99b1763f-5ca3-4321-aa1b-c58821eff167)
![Blender  C__Users_ryosu_Desktop_その他_一時保管フォルダ_Blender249 Accel World VS Sword Art Online PC mdl anm 2017-12-04 _Blender249 Accel World VS Sword Art Online PC mdl anm 2017-12-04 _Blender249 blend  2023_11_14 1_12_01](https://github.com/rbfx/rbfx/assets/148245940/4e3639af-e5a6-4683-bc3b-6b36ab8b6118)
![Blender  C__Users_ryosu_Desktop_その他_一時保管フォルダ_Blender249 Accel World VS Sword Art Online PC mdl anm 2017-12-04 _Blender249 Accel World VS Sword Art Online PC mdl anm 2017-12-04 _Blender249 blend  2023_11_14 1_13_22](https://github.com/rbfx/rbfx/assets/148245940/c4f88b27-80e1-4579-804d-586db2fd45b8)
![Blender  C__Users_ryosu_Desktop_その他_一時保管フォルダ_Blender249 Accel World VS Sword Art Online PC mdl anm 2017-12-04 _Blender249 Accel World VS Sword Art Online PC mdl anm 2017-12-04 _Blender249 blend  2023_11_14 1_13_52](https://github.com/rbfx/rbfx/assets/148245940/652d37fa-4140-4155-9e9b-8229de87e7a3)
![Blender  C__Users_ryosu_Desktop_その他_一時保管フォルダ_Blender249 Accel World VS Sword Art Online PC mdl anm 2017-12-04 _Blender249 Accel World VS Sword Art Online PC mdl anm 2017-12-04 _Blender249 blend  2023_11_14 1_13_09](https://github.com/rbfx/rbfx/assets/148245940/7ee478cc-8f30-4305-a5ff-6e558d672252)
![Blender  C__Users_ryosu_Desktop_その他_一時保管フォルダ_Blender249 Accel World VS Sword Art Online PC mdl anm 2017-12-04 _Blender249 Accel World VS Sword Art Online PC mdl anm 2017-12-04 _Blender249 blend  2023_11_14 1_08_26](https://github.com/rbfx/rbfx/assets/148245940/8d2d328b-7d6f-419a-8c00-bef2c44d9ef0)
![Blender  C__Users_ryosu_Desktop_その他_一時保管フォルダ_Blender249 Accel World VS Sword Art Online PC mdl anm 2017-12-04 _Blender249 Accel World VS Sword Art Online PC mdl anm 2017-12-04 _Blender249 blend  2023_11_14 1_12_38](https://github.com/rbfx/rbfx/assets/148245940/633a41ff-80fb-4d9c-b46a-639e1b4cd719)
![Blender  C__Users_ryosu_Desktop_その他_一時保管フォルダ_Blender249 Accel World VS Sword Art Online PC mdl anm 2017-12-04 _Blender249 Accel World VS Sword Art Online PC mdl anm 2017-12-04 _Blender249 blend  2023_11_14 1_13_37](https://github.com/rbfx/rbfx/assets/148245940/494791a0-4471-415c-9e49-628f9007df46)
![Blender  C__Users_ryosu_Desktop_その他_一時保管フォルダ_Blender249 Accel World VS Sword Art Online PC mdl anm 2017-12-04 _Blender249 Accel World VS Sword Art Online PC mdl anm 2017-12-04 _Blender249 blend  2023_11_14 1_14_04](https://github.com/rbfx/rbfx/assets/148245940/74c2dbdb-6333-4a0c-b64b-0b5303e94a7a)

Sorry for my late reply.

I will post a screenshot of the script.

**Reply by rokups** [reply]
I am sorry, but we can not help you, as your question is in no way related to our project.
